File: Code/new_fitervari_backend/src/main/java/at/htl/control/ExerciseHistoryRepository.java
package at.htl.control;

import at.htl.model.ExerciseHistory;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class ExerciseHistoryRepository implements PanacheRepository<ExerciseHistory> {

    public ExerciseHistory save(ExerciseHistory entity) {
        return getEntityManager().merge(entity);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/ExerciseHistoryRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.Customer
import at.htl_leonding.model.ExerciseHistory
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class ExerciseHistoryRepository : PanacheRepository<ExerciseHistory> {
    @Inject
    lateinit var em: EntityManager

    fun save(exerciseHistory: ExerciseHistory): ExerciseHistory {
        return em.merge(exerciseHistory)
    }
}